pages/authNotAllowed/authNotAllowed.html.js

```diff
@@ -0,0 +1,79 @@
+const { html } = require('@prairielearn/html');
+const { renderEjs } = require('@prairielearn/html-ejs');
+
+function getUrlForProvider(provider) {
+  switch (provider.name) {
+    case 'SAML':
+      return `/pl/auth/institution/${provider.institution_id}/saml/login`;
+    case 'Google':
+      return '/pl/oauth2login';
+    case 'Azure':
+      return '/pl/azure_login';
+    case 'Shibboleth':
+      return '/pl/shibcallback';
+    default:
+      return null;
+  }
+}
+
+function AuthNotAllowed({ institutionAuthnProviders, resLocals }) {
+  // LTI providers are special in that we don't currently allow an LTI session
+  // to be initiated from PrairieLearn. We'll handle them specially.
+  const hasLtiProvider = institutionAuthnProviders.some((p) => p.name === 'LTI');
+  const nonLtiAuthnProviders = institutionAuthnProviders.filter((p) => p.name !== 'LTI');
+  const hasNonLtiProviders = nonLtiAuthnProviders.length > 0;
+
+  // Check if there's a default authentication provider.
+  const defaultAuthnProvider = nonLtiAuthnProviders.find((p) => p.is_default);
+
+  return html`
+    <!DOCTYPE html>
+    <html lang="en">
+      <head>
+        ${renderEjs(__filename, "<%- include('../partials/head') %>", resLocals)}
+      </head>
+      <body>
+        ${renderEjs(__filename, "<%- include('../partials/navbar') %>", {
+          ...resLocals,
+          navbarType: 'unauthenticated',
+        })}
+        <div class="container">
+          <h1>Unsupported authentication method</h1>
+          <p>
+            The authentication method you tried to use is not supported by your institution.
+            ${hasNonLtiProviders ? 'Please use an alternative authentication method.' : ''}
+            ${!hasNonLtiProviders && !hasLtiProvider
+              ? 'Please contact your institution for more information.'
+              : ''}
+          </p>
+          ${defaultAuthnProvider
+            ? html`
+                <div class="mb-3">
+                  <a href="${getUrlForProvider(defaultAuthnProvider)}" class="btn btn-primary">
+                    Sign in with institution single sign-on
```

Comment:
Should we make this look identical to the button they should have clicked on originally, to train them which one is the right one? That is, if we wanted them to have originally click on "U of I" then this button should also say "U of I".

Reply:
Hmm, interesting point... taking a different approach, what if I just merged this error page into the login page via some query parameters, e.g. you'd get redirected to `prairielearn.org/pl/login?institution_id=1&unsupported_authn_provider=true` which then shows something like this (I'll obviously improve the copy/layout/etc.):

<img width="550" alt="Screen Shot 2022-08-22 at 14 01 39" src="https://user-images.githubusercontent.com/1476544/186018394-1b3500e8-44aa-45fa-8fb9-9d4055f1f494.png">

Reply:
I ultimately made this change!